### PR TITLE
Update javadoc for withValue(s) to indicate withValue can be chained

### PR DESCRIPTION
**Pull request by nathanhleung**
See #6507 for discussion

### CODE REVIEW

File: context/src/main/java/io/grpc/Context.java [hunk 1]
```diff
@@ -326,6 +326,19 @@ public CancellableContext withDeadline(Deadline newDeadline, ScheduledExecutorSe
    *   });
    * </pre>
    *
+   * Note that multiple calls to {@link #withValue} can be chained together.
```
Comment (ejona86) [inline context/src/main/java/io/grpc/Context.java]: Needs `<p>`. Ditto for "Nonetheless". See Travis failure.

**Comment by nathanhleung** [comment]
Oops, thanks -- fixed and rebased

**Comment by ran-su** [comment]
@nathanhleung Merged. Thanks.

File: context/src/main/java/io/grpc/Context.java [hunk 2]
```diff
@@ -326,6 +326,19 @@ public CancellableContext withDeadline(Deadline newDeadline, ScheduledExecutorSe
...
    * </pre>
    *
+   * <p>Note that multiple calls to {@link #withValue} can be chained together.
+   * That is,
+   *
+   * <pre>
+   * context.withValues(K1, V1, K2, V2);
+   * // is the same as
+   * context.withValue(K1, V1).withValue(K2, V2);
+   * </pre>
```
Comment (bogdandrutu) [inline context/src/main/java/io/grpc/Context.java]: @ejona86 I thought that the first is more performant, if that is the case we should add that to the comment.

Comment (ejona86) [reply context/src/main/java/io/grpc/Context.java]: The memory/performance difference isn't as much now that we have the HashArrayMappedTrie. We had different code paths when we used the linked list. With the new implementation values are packed together automatically, and the double-value version just does two adds internally:

https://github.com/grpc/grpc-java/blob/a8bdf5cb1abf7a857e263a5d1500f1bba2c940e3/context/src/main/java/io/grpc/Context.java#L354

The only difference between the two would be an additional Context allocation. Most users shouldn't concern themselves with that difference.

**Pull request merged**